core/mo-services-impl/nmf-platform-generic-impl/src/main/java/esa/mo/platform/impl/provider/gen/GPSProviderServiceImpl.java

```diff
@@ -321,36 +321,71 @@ public GetLastKnownPositionResponse getLastKnownPosition(MALInteraction interact
   @Override
   public void getPosition(GetPositionInteraction interaction)
       throws MALInteractionException, MALException
+  {
+    boolean useTLEPropagationResults[] = useTLEPropagation();
+    boolean useTLEpropagation = useTLEPropagationResults[0];
+    boolean throwException = useTLEPropagationResults[1];  
+
+    if (throwException){
+      throw new MALInteractionException(
+          new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
+    }
+
+    interaction.sendAcknowledgement();
+
+    Position position = updateCurrentPosition(useTLEpropagation);
+    if (position == null) {
+      throw new MALInteractionException(
+          new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
+    }
+    interaction.sendResponse(position);
+  }
+
+  /**
+   * Checks if TLE propagation should be used
+   * and enables/disables a throwException flag
+   * @return useTLEpropagation result[0]
+   *         throwException result[1]
+   */
+
+  public boolean[] useTLEPropagation()
   {
     boolean useTLEpropagation = false;
+    boolean throwException = false;
```

Comment:
Why not throw an exception in this method already and then catch/rethrow/ignore it if needed? Using bool flag to indicate that a caller should return exception is very confusing.